src/grt/src/fastroute/src/FastRoute.cpp

```diff
@@ -578,17 +579,20 @@ void FastRouteCore::initBlockedIntervals(std::vector<int>& track_space)
     int edge_cap = getEdgeCapacity(x, y, x, y + 1, layer);
     if (edge_cap > 0) {
       int reduce = 0;
-      for (const auto& interval_it : intervals) {
-        reduce += ceil(static_cast<float>(
-                           std::abs(interval_it.upper() - interval_it.lower()))
-                       / track_space[layer - 1]);
+      if (layer > 0 && layer <= track_space.size()) {
```

Comment:
I see that for the design shared in issue https://github.com/The-OpenROAD-Project/OpenROAD/issues/5574 it is trying to access track_space vector with indexes (layer) 8 for the horizontal loop and 9 for the vertical loop, while track_space has a size of 7.

Reply:
I see this layers on the gui, so I guess track_space is wrong with a size of only 7?
![image](https://github.com/user-attachments/assets/cd4b4300-5d32-4683-b3dc-7de775450a7f)

Reply:
What is the layer range defined in the grt script?

Reply:
In vars file we have:
`export MIN_ROUTING_LAYER="M2"
export MAX_ROUTING_LAYER="M7"`

In /flow/scripts/global_route.tcl we do not seem to define a layer range, we only use it for the layer adjustments.

Reply:
maybe you meant to print them, I will do that

Reply:
No, this is what I was thinking. The layer range is only until M7, so M8 and M9 obstructions should be ignored. You need to figure out where we are inserting these obs outside the layer range.

Reply:
Ok, will do. I guess the condition I added is healthy to stay either way, right?

Reply:
It is, but I think it should throw an error if that happens. We have many checks before reaching this function, so if we pass something out of the layer range, we should error out. This will help to identify other possible errors in the GlobalRouter class.

Reply:
I did not notice Matt's comment before, but it seems he is right, fastroute looks to be being initialized before the min and max routing layers are set on the gui.tcl.

Reply:
I see some calls to `findInstancesObstructions()` and `findNetsObstructions()` happening before min and max are properly set, they have 1 and 10, this happens during `log_cmd estimate_parasitics -global_routing` from the gui.tcl. 

After gui.tcl finishes and I click on rudy heatmap, initFastRoute() has the correct min and max with 2 and 7. Maybe grt was initialized previously with the wrong min and max?

Reply:
@eder-matheus perhaps we need to store the range in odb so that it is available at startup.

Reply:
> @eder-matheus perhaps we need to store the range in odb so that it is available at startup.

Agreed! I believe @gudeh had a PR that was closed in the past. Perhaps we can restore it.

Reply:
> Agreed! I believe @gudeh had a PR that was closed in the past. Perhaps we can restore it.

Nvmd, the PR I was thinking of relocates the layer adjustments to odb.